telemetry_package/main.py

```diff
@@ -1,33 +1,145 @@
+
+            # After connecting, create two subscriptions: one to the '/rosout' topic and one to the 'control' topic
+            self.rosout_subscription = self.create_subscription(
+                Log,
+                '/rosout',
+                self.rosout_callback, 
+                qos_profile = qos_profile)
+            
+            self.rosout_subscription
+            print('Created subscription to "/rosout".')
+
+            # Commenting this out for test purposes
+            # self.control_subscription = self.create_subscription(
+            #     Log,
+            #     'control',
+            #     self.control_subscription_callback, 
+            #     qos_profile = qos_profile)
+
+            # self.control_subscription 
+            print('Created subscription to "control".')
+
+
+            self.heartbeat_subscription = self.create_subscription(
+               Log,
+               'heartbeat', 
+                qos_profile = qos_profile)
+            
+            self.heartbeat_subscription
+            print('created subscription to "heartbeat".')
+
+            
+
+        except Exception as e:
+            print(f'Error occurred: {e}')
+            self.server_sock.close()
+            os.remove(DEFAULT_UNIX_SOCKET_PATH)
+        
+
+    '''
+    @brief This function sanitizes incoming messages so that they do not collide with JSON format guidelines
+
+    @param msg message to sanitize
+
+    @return str
+    '''
+    def sanitize_message(self, msg : str)->str:
+        try: 
+            msg = re.sub("[\{\}:']", "", msg) 
+            return msg 
+        except Exception as e: 
+            print(f'Error occurred during string sanitization: {e}')
+
+
+    '''
+    @brief callback of the subcriptions. 
+
+    This function takes the content of a log message from the /rosout topic, sanitizes it (altough still quite dirtily) and sends it to the ground station webapp via UNIX socket. 
+
+    @param log_msg the message passed by the topic
+    '''
+    def rosout_callback(self, log_msg)->None:
+        try: 
+            log_msg.msg = self.sanitize_message(log_msg.msg)
+
+            #@di-math suggests that this change (#b"\x17") fixes the JSON strings being stitched togeher error 
+            self.client_sock.sendall(json.dumps({"type": "std" , "content" :  log_msg.msg}).encode() + b"\x17")
+        except Exception as e:
+            print(f'Error occurred in rosout_callback: {e}')
+
+
+    '''
+    @brief callback of the subcriptions. 
+
+    This function takes the content of a message from the control topic sanitizes it (altough still quite dirtily) and sends it to the ground station webapp via UNIX socket. 
+
+    @param msg the message passed by the topic
+    '''
+    def control_callback(self, msg)->None:
+        try: 
+            msg.data = self.sanitize_message(msg.data)
+            self.client_sock.sendall(json.dumps({"type": "std" , "content" :  msg.data}).encode() + b"\x17")
```

Comment:
Let's have the following as JSON representation of the Heartbeat msg_type : 
```typescript
{ "type": "heartbeat"
  "content":  {
         "sender_id": string,
         "tick": int,
         "active": Boolean,
         "timestamp": string
         }
}
```

Reply:
This dictionary is supposed to refer to the Control msg_type. Did you misstype? Or do you want me to send the heartbeat to the ground-station webapp as well?

Reply:
As far as I know, @di-math only wants you to send the heartbeat and not control (which is also kind of included in the heartbeat anyway).